test_app/tests/lib/utils/test_views.py

```diff
@@ -65,6 +66,11 @@ def test_ansible_base_view_parent_view(caplog, setting, log_message, default_par
             assert log_message in caplog.text
 
 
+def is_py_311():
+    return bool(sys.version.startswith('3.11'))
+
+
+@pytest.mark.xfail(reason='Type checking calls import_module', condition=is_py_311)
```

Comment:
This seems like we're losing this test on 3.11 then? I'm not really a fan of that. Can we just skip type checking for this function on 3.11 instead?

Reply:
It's more complicated than that. Assume type checking is already turned off. That doesn't prevent `typeguard` _logic_ from running. The problem is that this mocks stuff from `importlib` which is commonly called by other libraries... libraries like `typeguard`. Turning off enforcement doesn't mean disconnecting its logic.

It is hard to co-exist with `mock.patch('importlib.import_module', side_effect=ImportError("Test Exception"))`. And if you wanted a long-term durable solution, I would add another method and mock that method instead of something in `importlib`. I hesitated to go that far here because I didn't want to change the code itself.

Reply:
I didn't want to do this because it goes so far out of my intended scope of this PR, but I introduced a new utility method so that the test doesn't mock `importlib.import_module` because _it shouldn't do this_.

Ping @john-westcott-iv as well. Since _my last rebase_ more incorrect type hints have been added. I have no issue whatsoever with people adding code with no type hints. What I do not like is code with provably wrong type hints. I've gotten rid of the hack here so that hopefully there should be no more potential blockers. I legitimately have a use case with RBAC, I would add more type hints for the more complex contracts but without regression coverage it's a non-starter.

Reply:
How are you checking the type hints? I'm wondering if we can create a GitHub action to enforce it on PRs and/or add a pre-commit hook?

Reply:
@john-westcott-iv this PR checks type hints at runtime via tests. It is integrated into tox testing, so this enforces type hint correctness (at least for those which are tested) as a CI gate.

Reply:
Perfect, we won't need a seperate action because it will just be part of the tox tests.